### ISSUE TITLE
Feature Request: Users should not see TLP Group if they are not in the group

### ISSUE THREAD

**Issue by ociappara** [CybercentreCanada/assemblyline#90]
**Is your feature request related to a problem? Please describe.**

When submitting a sample to Assemblyline users can select a TLP and a group. Users who are not assigned to a group should not see the group in the UI when making a submission. 

Example: System has the below context

    Team AR 
    Team AT
    Team BR
    Team BT

User X is Assigned **Team AR**, when User X tries to submit a file they are able to see AR, AT, BR, BT if the user picks any of AT, BR, BT an error shows up, because the user is not assigned that context. 

To allow for an easier user experience the UI should only show Groups that that user is assigned.

**Comment by cccs-sgaron** [comment]
The reason it is done this way is to be able to share analysis between groups. 

Ex: I'm in team AR and I want to share the analysis with team BT. Therefore I should be able to select AR and BT. Yes you are right, this can lead to selecting a combinaison where it is actually invalid for you... 

Maybe a compromise would be to have the UI show you in advance that this classification is invalid for you instead of waiting for the submission to be sent?

**Comment by ociappara** [comment]
Yes a UI element showing this would be a nice way to handle this.

Not sure about one thing, I will test it out, but don't you need to be in both AR and BT groups to be able to select this combination? if that's the case removing groups you're not part of in the UI should not impact this behaviour.

**Comment by cccs-sgaron** [comment]
I think this issue here comes down to the wording that we chose for the classification.

**Levels** is basically a number from 0 to inf where if you have a level you get access to all levels below
**Required** is a list of tokens that you must have to get access to the data (uses an AND operator)
**Groups** and **Sub-Groups** is a list of token that you must be member of at least one of to get access to the data (uses an OR operator)

So in your case, if you want to hide what you call groups from the ppl that don't have access to it. You can put your group definitions in the required section and a user would have to be part of all mentioned groups to get access and see that the group exists.

If you use the group section, a user has to be only a member of one of the selected group to get access to the data. 

**WARNING**: Moving your groups into the required section is not a trivial thing because it will render all old data unaccessible because those part are broken down into multiple field in the DB to compute proper access control using filter queries.

I can still proceed witht the UI warning for now as I added classification checking in the UI already for the external links
